packages/http/src/validator/__tests__/index.spec.ts

```diff
@@ -53,30 +53,61 @@ describe('HttpValidator', () => {
             input: { method: 'get', url: { path: '/' } },
             config: { mock: { dynamic: false }, validate: { request: { body: true } } },
           }),
-        ).toEqual([mockError]);
+        ).toEqual([]);
 
         expect(resolveValidationConfigModule.resolveRequestValidationConfig).toHaveBeenCalled();
-        expect(httpBodyValidator.validate).toHaveBeenCalledWith(undefined, [], undefined);
+        expect(httpBodyValidator.validate).not.toHaveBeenCalled();
         expect(httpHeadersValidator.validate).not.toHaveBeenCalled();
         expect(httpQueryValidator.validate).not.toHaveBeenCalled();
       };
 
       describe('request is not set', () => {
-        it('validates body', test());
+        it('does not try to validate the body', test());
       });
 
       describe('request is set', () => {
         describe('request.body is not set', () => {
-          it('validates body', test({ request: { path: [], headers: [], query: [], cookie: [] } }));
+          it('does not try to validate the body', test({ request: { path: [], headers: [], query: [], cookie: [] } }));
         });
 
         describe('request.body is set', () => {
-          it(
-            'validates body',
-            test({
-              request: { body: { contents: [] }, path: [], query: [], headers: [], cookie: [] },
-            }),
-          );
+          describe('request body is not required', () => {
```

Comment:
This is a good use of describe but I think you could improve the reliability and readability of these tests. It looks like your setup was copied and pasted from the setup in the test() method, which means that method could change, and then the test is not actually confirming the difference between required true/false m, but confirming the difference between test and this copied and pasted code. 

Can we either pass in some arguments for request object; or find a less hidden approach? 

Often setting up an object with the default structure, then overriding specific parts in a single test, is a good way to make sure you’re really testing the difference behaviour related to that change.

Reply:
Yeah I think this is a good point, I'll refactor this and make sure it's better

Reply:
Done in ecee804